llpc/translator/lib/SPIRV/SPIRVReader.cpp

```diff
@@ -1853,7 +1853,7 @@ Constant *SPIRVToLLVM::buildConstStoreRecursively(SPIRVType *const spvType, Type
                                      constStoreValue->getAggregateElement(i));
 
       if (needsPad)
-        constElements[i] = ConstantExpr::getInsertValue(constElements[i], constElement, 0);
+        constElements[i] = static_cast<Constant *>(getBuilder()->CreateInsertValue(constElements[i], constElement, 0));
```

Comment:
Never use `static_cast` for this kind of thing, use `cast` instead (it has a safety check).

Reply:
If the cost of the check is a problem, we should build without assertions. It's too easy to get these casts wrong otherwise.

ConstantFoldInsertValueInstruction is allowed to fail and return `nullptr`, which isn't great.

Reply:
That's a fair point.